controllers/serviceexport/controller_test.go

```diff
@@ -22,10 +21,25 @@ func TestNewController(t *testing.T) {
 	memberDynamicInformerFactory := dynamicinformer.NewDynamicSharedInformerFactory(memberDynamicClient, time.Second*30)
 	hubDynamicInformerFactory := dynamicinformer.NewDynamicSharedInformerFactory(hubDynamicClient, time.Second*30)
 
-	_, err := serviceexport.New("fake-cluster", memberKubeClient, memberDynamicClient, hubDynamicClient,
+	fakeClusterID := "fake-cluster"
+
+	c, err := New(fakeClusterID, memberKubeClient, memberDynamicClient, hubDynamicClient,
 		memberSharedInformerFactory, memberDynamicInformerFactory, hubDynamicInformerFactory)
 
 	if err != nil {
 		t.Errorf("failed to create a new serviceexport controller: %v", err)
 	}
+
+	if c.memberClusterID != fakeClusterID {
```

Comment:
please compare the full structure, instead of one by one. You could use [cmpopts.IgnoreInterfaces](https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#IgnoreInterfaces) to ignore the fields.

https://github.com/golang/go/wiki/TestComments#compare-full-structures

Reply:
Like I've mentioned earlier, the only comparable field (after ignoring the interfaces) would be the memberClusterID field (a string). Plus, Controller struct does not have any exported field.

Reply:
I did not notice that there were no exported fields. Then it should be fine.